share/janitor/tasks.py

```diff
@@ -2,19 +2,40 @@
 
 import celery
 
+from django.db.models import Exists
+from django.db.models import OuterRef
+
 from share import tasks
 from share.models import RawDatum
+from share.models import NormalizedData
 
 
 logger = logging.getLogger(__name__)
 
 
 @celery.shared_task(bind=True)
-def rawdata_janitor(self, limit=100):
+def rawdata_janitor(self, limit=500):
     """Find RawDatum that do not have a NormalizedData process them
     """
     count = 0
-    for rd in RawDatum.objects.select_related('suid__source_config').filter(normalizeddata__isnull=True).order_by('id')[:limit].iterator():
+
+    # NOTE: Do NOT use .iterator here. It will create a temporary table and eat disk space like no other
+    # the limit lets this query fit nicely in memory and actually finish executing
+    # Be very careful about changing this query. If you do change it, make sure the EXPLAIN looks something like this:
```

Comment:
I trimmed the original explain to remove the parts that aren't super important.
Really the only important part is `Nested Loop Anti Join`.
Postgres' query planner will decide the optimal way to perform a query based on the available hardware, data size, etc, etc. Given that you only have ~500 rows a join is probably faster.